ios/PurchasesHybridCommon/PurchasesHybridCommon/PaywallProxy.swift

```diff
@@ -27,53 +27,65 @@ import UIKit
     }
 
     @objc
     public func presentPaywall(displayCloseButton: Bool) {
-        presentPaywall(optionalDisplayCloseButton: displayCloseButton)
+        presentPaywall(optionalDisplayCloseButton: displayCloseButton, optionalOffering: nil)
+    }
+
+    @objc
+    public func presentPaywall(offering: Offering) {
+        presentPaywall(optionalDisplayCloseButton: nil, optionalOffering: offering)
+    }
+
+    @objc
+    public func presentPaywall(offering: Offering, displayCloseButton: Bool) {
+        presentPaywall(optionalDisplayCloseButton: displayCloseButton, optionalOffering: offering)
     }
 
     @objc
     public func presentPaywallIfNeeded(requiredEntitlementIdentifier: String) {
         presentPaywallIfNeeded(requiredEntitlementIdentifier: requiredEntitlementIdentifier,
-                               optionalDisplayCloseButton: nil)
+                               optionalDisplayCloseButton: nil,
+                               optionalOffering: nil)
     }
 
     @objc
-    public func presentPaywallIfNeeded(requiredEntitlementIdentifier: String, displayCloseButton: Bool) {
+    public func presentPaywallIfNeeded(requiredEntitlementIdentifier: String, 
+                                       displayCloseButton: Bool) {
         presentPaywallIfNeeded(requiredEntitlementIdentifier: requiredEntitlementIdentifier,
-                               optionalDisplayCloseButton: displayCloseButton)
+                               optionalDisplayCloseButton: displayCloseButton,
+                               optionalOffering: nil)
     }
 
-    private func presentPaywallIfNeeded(requiredEntitlementIdentifier: String, optionalDisplayCloseButton: Bool?) {
+    private func presentPaywallIfNeeded(requiredEntitlementIdentifier: String, 
+                                        optionalDisplayCloseButton: Bool?,
+                                        optionalOffering: Offering?) {
         _ = Task { @MainActor in
             do {
                 let customerInfo = try await Purchases.shared.customerInfo()
                 if !customerInfo.entitlements.active.keys.contains(requiredEntitlementIdentifier) {
-                    if let displayCloseButton = optionalDisplayCloseButton {
-                        self.presentPaywall(displayCloseButton: displayCloseButton)
-                    } else {
-                        self.presentPaywall()
-                    }
-
+                    self.presentPaywall(optionalDisplayCloseButton: optionalDisplayCloseButton,
+                                        optionalOffering: optionalOffering)
                 }
             } catch {
                 NSLog("Failed presenting paywall: \(error)")
             }
         }
     }
 
-    private func presentPaywall(optionalDisplayCloseButton: Bool?) {
+    private func presentPaywall(optionalDisplayCloseButton: Bool?,
+                                optionalOffering: Offering?) {
```

Comment:
Nit: this is against the Swift guidelines, optional shouldn't be in the name:
```suggestion
    private func presentPaywall(displayCloseButton: Bool?,
                                offering: Offering?) {
```

Reply:
What would be a good naming so it doesn't colide with the other `presentPaywall` function? That one doesn't have optionals as parameters since it's `@objc`